src/App.tsx

```diff
@@ -1,39 +1,71 @@
-import React from 'react';
+import { Component } from 'react';
+
+import { Clock } from './components/Clock';
+
 import './App.scss';
 
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timer = 0;
+
+  componentDidMount() {
+    this.timer = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+      this.setState({ hasClock: false });
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('click', (event) => {
+      event.preventDefault();
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+      this.setState({ hasClock: true });
+    });
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    clearInterval(this.timer);
 
-        {' time is '}
+    document.removeEventListener('contextmenu', (event) => {
+      event.preventDefault();
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+      this.setState({ hasClock: false });
+    });
+
+    document.removeEventListener('click', (event) => {
+      event.preventDefault();
+
+      this.setState({ hasClock: true });
+    });
+  }
```

Comment:
I wouldn't work)
You have to pass a link to the function, so it will be the same and method removeEventListener will understand wich function you want to remove. BUT now you have passed one function in addEventListener and different one in removeEventListener

![image](https://user-images.githubusercontent.com/122989585/230707072-e30068fe-cfab-40c5-bc96-1efff322bb5d.png)

Reply:
create separated methods in the class and use them